### PR TITLE
Substrate Updates

### PR DESCRIPTION
**Pull request by AurevoirXavier**
https://github.com/darwinia-network/substrate-update-tracking/issues/2

### CODE REVIEW

**Comment by HackFisher** [comment]
https://github.com/paritytech/substrate/blob/master/docs/Upgrading-2.0-to-3.0.md

**Review by HackFisher — COMMENTED** [review]
Migrations need to take care:
https://github.com/darwinia-network/darwinia-common/pull/468/files#diff-194656be9ce9133b248a5b7cd18842eb5d3faacde3f1221b32e654419870136aR109

**Comment by AurevoirXavier** [comment]
@AsceticBear Please fix the EVM test

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -44,9 +44,9 @@ jobs:
 
       - name: Build and Run node
         run: |
-            cargo b
-            cargo r -- --dev --tmp &
-            sleep 60
+          cargo build
```
Comment (AsceticBear) [inline .github/workflows/ci.yml:47]: `cargo build --release` will be faster for the compile process.  The default build directory is `target/debug`

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -22,13 +23,13 @@ jobs:
...
         with:
-          toolchain: nightly-2020-09-20
+          toolchain: nightly
           override: true
           target: wasm32-unknown-unknown
 
       - name: Overall test
-        run: cargo test
-      
+        run: cargo t
```
Comment (AsceticBear) [inline .github/workflows/ci.yml:31]: `cargo test`?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (AurevoirXavier) [reply .github/workflows/ci.yml:47]: `--release` will set the opt level to 3

much slower

**Comment by HackFisher** [comment]
Need to check transaction version bump:

File: frame/dvm/src/mock.rs [hunk 3]
```diff
@@ -196,7 +191,7 @@ pub struct AccountInfo {
 }
 
 fn address_build(seed: u8) -> AccountInfo {
-	let private_key = H256::from_slice(&[(seed + 1) as u8; 32]); //H256::from_low_u64_be((i + 1) as u64);
+	let private_key = H256::from_slice(&[(seed + 1) as u8; 32]); //H256::from_lowdarwinia_balances::GenesisConfig::<Test, RingInstance> + 1) as u64);
```
Comment (AsceticBear) [inline frame/dvm/src/mock.rs:194]: The changes in this line should be rolled back

**Review by AsceticBear — CHANGES_REQUESTED** [review]

**Comment by HackFisher** [comment]
@AurevoirXavier @AsceticBear There are conflicts after #490 being merged.

**Comment by HackFisher** [comment]
https://github.com/darwinia-network/darwinia-common/pull/468/files#diff-847a61d013ce4aedef41be8e70a2b09d96eb3ac5d5ce3d3da45e03915ab1fa90R1040

**Review by HackFisher — APPROVED** [review]

**Comment by HackFisher** [comment]
@AsceticBear Please help review changes requested.

**Review by AsceticBear — APPROVED** [review]

**Pull request merged**